web/src/main/java/org/cbioportal/web/parameter/filter/PatientTreatmentFilter.java

```diff
@@ -0,0 +1,29 @@
+package org.cbioportal.web.parameter.filter;
+
+import org.cbioportal.model.PatientTreatmentRow;
+import org.cbioportal.web.parameter.SampleIdentifier;
+
+import java.util.Map;
+
+public class PatientTreatmentFilter {
+    private String treatment;
+
+    /**
+     * A sampleId should be included if the treatment row that corresponds
+     * to the treatment and time in this filter contains that sampleId.
+     * @param sampleId sample.STABLE_ID
+     * @param treatments key is PatientTreatmentRow::calculateKey
+     */
+    public boolean filter(SampleIdentifier sampleId, Map<String, PatientTreatmentRow> treatments) {
+        PatientTreatmentRow row = treatments.get(treatment);
+        return row != null && row.getSamples().contains(sampleId.getSampleId());
```

Comment:
I think check for sample in TreatmentRow samples list might be an expensive operation(if there are many samples and/or if there are many treatments in selection). And there can be same sample id for multiple studies. I think you need to check for both sampleId and studyId together.

I propose to change replace sample and studies in both `PatientTreatmentRow` and `SampleTreatmentRow` to a `Map<SampleIdentifier, Boolean>` if not to a List<SampleIdentifier> -> to support duplicate sampleIds.

This goes same with `SampleTreatmentFilter.java`